### DISCUSSION TITLE
How do the translations work?

### DISCUSSION THREAD

**Discussion by JavaJeremy** [ReceiptManager/Application#26] [Q&A] [answered]
I wanted to improve some of the German in the app, but I became confused because there were multiple places where the same sentences were at. 
This lead me to thinking: Is there a central place where the translations originate from and the other files are just generated, or are there different files for different devices (like for tablets/webapps) that I should test the translation with first?

Here's an example:
![grafik](https://user-images.githubusercontent.com/7582225/115076487-eb350600-9efc-11eb-9bf2-ad4fb55f527f.png)
When wanting to change "Hinzufügen eines API token" I didn't know if I have to change this in all 3 files, or not.

**Comment by monolidth** [comment, marked as answer]
> I wanted to improve some of the German in the app, but I became confused because there were multiple places where the same sentences were at.

The reason why this sentence appear on multiple "places" is the scope e.g. the label and the hint text are the same but the current version gives me the opportunity to change both independent. Therefore, you have to change both but wait:

* I currently refactor the application and I removed the localisation (at the moment) because I want to reduce unused language string and therefore, I'm not using the language files.

* You could help me, if you translate the English language files but I have to create them first :)

Regards,

William

**Reply by JavaJeremy** [reply]
okay, good to know - then I'll wait and won't touch the current version 😄 
But since around 90% of the current German translation is already perfect from my point of view, retranslating it from scratch would prbly be a bit of an overkill^^

**Reply by monolidth** [reply]
I now support the German language. You can try it out now. If you have any issues, please let me know.
